// nativescript-facebook/login-manager.ios.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var login_response_1 = require("./login-response");
var FB_LOGIN_BEHAVIOUR = 2;
var LOGIN_PERMISSIONS = ["public_profile", "email"];
var loginManager;
function init(fbId) {
    setAppId(fbId);
    loginManager = FBSDKLoginManager.alloc().init();
    loginManager.loginBehavior = FB_LOGIN_BEHAVIOUR;
}
exports.init = init;
function _registerLogoutCallback(callback) {
    exports.onLogoutCallback = callback;
}
exports._registerLogoutCallback = _registerLogoutCallback;
function _registerLoginCallback(callback) {
    exports.onLoginCallback = function (result, error) {
        if (error) {
            callback(new Error(error.localizedDescription));
            return;
        }
        if (!result) {
            callback(new Error("Fatal error"));
            return;
        }
        if (result.isCancelled) {
            callback(new Error('canceled'));
            return;
        }
        if (result.token) {
            var token = result.token.tokenString;
            var loginResponse = new login_response_1.LoginResponse(token);
            callback(null, loginResponse);
        }
        else {
            callback(new Error("Could not acquire an access token"));
            return;
        }
    };
}
exports._registerLoginCallback = _registerLoginCallback;
function setAppId(fbAppId) {
    FBSDKSettings.setAppID(fbAppId);
}
function requestPublishPermissions(permissions, callback) {
    _registerLoginCallback(callback);
    loginManager.logInWithPublishPermissionsHandler(permissions, exports.onLoginCallback);
}
exports.requestPublishPermissions = requestPublishPermissions;
function requestReadPermissions(permissions, callback) {
    _registerLoginCallback(callback);
    loginManager.logInWithReadPermissionsHandler(permissions, exports.onLoginCallback);
}
exports.requestReadPermissions = requestReadPermissions;
function login(callback) {
    requestReadPermissions(LOGIN_PERMISSIONS, callback);
}
exports.login = login;
function logout(callback) {
    loginManager.logOut();
    if (callback) {
        callback();
    }
}
exports.logout = logout;
